api/v1alpha1/iochaos_types.go

```diff
@@ -81,42 +52,27 @@ type IoChaosSpec struct {
 	// +optional
 	Value string `json:"value"`
 
-	// Duration represents the duration of the chaos action.
-	// It is required when the action is `PodFailureAction`.
-	// A duration string is a possibly signed sequence of
-	// decimal numbers, each with optional fraction and a unit suffix,
-	// such as "300ms", "-1.5h" or "2h45m".
-	// Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
-	// +optional
-	Duration *string `json:"duration,omitempty"`
-
-	// Layer represents the layer of the I/O action.
-	// Supported value: fs.
-	// Default layer: fs
-	// +kubebuilder:validation:Enum=fs
-	Layer IOLayer `json:"layer"`
+	// Action defines the specific pod chaos action.
+	// Supported action: latency / fault / attrOverride
+	// +kubebuilder:validation:Enum=latency;fault;attrOverride
+	Action IoChaosType `json:"action"`
 
 	// Delay defines the value of I/O chaos action delay.
 	// A delay string is a possibly signed sequence of
 	// decimal numbers, each with optional fraction and a unit suffix,
 	// such as "300ms".
 	// Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
-	//
-	// If `Delay` is empty, the operator will generate a value for it randomly.
 	// +optional
 	Delay string `json:"delay,omitempty"`
 
 	// Errno defines the error code that returned by I/O action.
 	// refer to: https://www-numi.fnal.gov/offline_software/srt_public_context/WebDocs/Errors/unix_system_errors.html
-	//
-	// If `Errno` is empty, the operator will generate an error code for it randomly.
 	// +optional
-	Errno string `json:"errno,omitempty"`
+	Errno uint32 `json:"errno,omitempty"`
 
-	// Percent defines the percentage of injection errors and provides a number from 0-100.
-	// default: 100.
+	// Attr defines the overrided attribution
 	// +optional
-	Percent string `json:"percent,omitempty"`
+	Attr *AttrOverrideSpec `json:"attr,omitempty"`
```

Comment:
@YangKeao Seems there is also a new field, need docs to describe it.

Reply:
I have described it in [document](https://github.com/chaos-mesh/chaos-mesh/blob/f117548019f6babf1be03bd185c27ba12704e537/website/docs/user_guides/io_chaos.md). It's a [struct](https://github.com/chaos-mesh/chaos-mesh/blob/f117548019f6babf1be03bd185c27ba12704e537/api/v1alpha1/podiochaos_types.go#L96) in go and a map in yaml (with limited keys)

Reply:
Got it.